pandas/core/index.py

```diff
@@ -123,14 +126,23 @@ def __new__(cls, data=None, dtype=None, copy=False, name=None, fastpath=False,
         if name is None and hasattr(data, 'name'):
             name = data.name
 
-        # no class inference!
         if fastpath:
             return cls._simple_new(data, name)
 
+        # range
+        if com.is_integer(data) and com.is_integer(dtype):
+            if copy is False:
+                copy = None
+            return RangeIndex(start=data, stop=dtype, step=copy, name=name)
```

Comment:
This looks very strange to me. I would not support this use of the generic `Index` constructor.